packages/sw-precaching/src/lib/models/precache-entries/object-precache-entry.js

```diff
@@ -1,43 +1,43 @@
-import ErrorFactory from '../error-factory';
-import RevisionedCacheEntry from './revisioned-cache-entry.js';
+import ErrorFactory from '../../error-factory';
+import RevisionedCacheEntry from './base-precache-entry.js';
 
 class DefaultsCacheEntry extends RevisionedCacheEntry {
-  constructor({revisionID, revision, url, cacheBust}) {
+  constructor({entryID, revision, url, cacheBust}) {
     // Check cacheBust is a boolean
     let filteredCacheBust = cacheBust;
     if (typeof filteredCacheBust === 'undefined') {
       filteredCacheBust = true;
     } else if (typeof filteredCacheBust !== 'boolean') {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad cacheBust Parameter. It should be a boolean value: ' +
           JSON.stringify(cacheBust)));
     }
 
     // Check revision is a string with a length
     if (typeof revision !== 'string' || revision.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad revision Parameter. It should be a string with at ' +
           'least one character: ' + JSON.stringify(revision)));
     }
 
     // Check url is a string.
     if (typeof url !== 'string' || url.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
+      throw ErrorFactory.createError('invalid-revisioned-entry',
         new Error('Bad url Parameter. It should be a string:' +
           JSON.stringify(url)));
     }
 
-    // Check revisionID type
-    if (typeof revisionID === 'undefined') {
-      revisionID = new URL(url, location.origin).toString();
-    } else if (typeof revisionID !== 'string' || revisionID.length === 0) {
-      throw ErrorFactory.createError('invalid-file-manifest-entry',
-        new Error('Bad revisionID Parameter. It should be a string with at ' +
-          'least one character: ' + JSON.stringify(revisionID)));
+    // Check entryID type
+    if (typeof entryID === 'undefined') {
+      entryID = new URL(url, location.origin).toString();
```

Comment:
You're correct in summing out what the behavior would be.

It's really a toss-up as to whether the SW location or the scope is a better base to use when there's a relative URL.

This is only an issue when a user explicitly changes the default scope (which usually isn't needed), and can be remediated by using absolute URLs for the assets, so I'm not too worried about it.